nexus/src/app/instance_platform.rs

```diff
@@ -492,3 +494,116 @@ impl super::Nexus {
         Ok(VmmSpec(spec))
     }
 }
+
+/// Yields the CPUID configuration to use for a VMM that specifies the supplied
+/// CPU `platform`.
+//
+// This is a free function (and not an `Into` impl on `VmmCpuPlatform`) to keep
+// all of the gnarly CPUID details out of the DB model crate, which defines that
+// type.
+fn cpuid_from_vmm_cpu_platform(
+    platform: db::model::VmmCpuPlatform,
+) -> Option<Cpuid> {
+    macro_rules! cpuid_leaf {
+        ($leaf:literal, $eax:literal, $ebx:literal, $ecx:literal, $edx:literal) => {
+            CpuidEntry {
+                leaf: $leaf,
+                subleaf: None,
+                eax: $eax,
+                ebx: $ebx,
+                ecx: $ecx,
+                edx: $edx,
+            }
+        };
+    }
```

Comment:
to be honest, i'm...not sure how i feel about the macro. i'd kiiiiinda rather see `eax: 0x..., ebx: 0x...` and so on, but maybe that ends up linewrapping and looking bad... :woman_shrugging: